src/main/java/io/vlingo/common/completes/AndThen.java

```diff
@@ -0,0 +1,39 @@
+package io.vlingo.common.completes;
+
+import java.util.function.Function;
+
+public class AndThen<I, O, NO> implements Operation<I, O, NO> {
+    private final Function<I, O> mapper;
+    private final O failedOutcome;
+    private Operation<O, NO, ?> nextOperation;
+
+    public AndThen(Function<I, O> mapper, O failedOutcome) {
+        this.mapper = mapper;
+        this.failedOutcome = failedOutcome;
+    }
+
+    @Override
+    public void onOutcome(I outcome) {
+        O next = mapper.apply(outcome);
+        if (next == failedOutcome) {
+            nextOperation.onFailure(next);
+        } else {
+            nextOperation.onOutcome(next);
+        }
+    }
+
+    @Override
+    public void onFailure(I outcome) {
+        nextOperation.onFailure((O) outcome);
```

Comment:
is there any relation in between I and O that allow this to be castable?

Reply:
AndThen can not map failures, so the type of the failure will remain the same between operations. For changing the type of the failure, you will need to use otherwise or recover. That's why here I = O (but I agree it's quite obscure). Maybe there is a way to make it more explicit, I will take a look.

Reply:
can it be said that both extend from the same type?